cfg/builder/builder_walk.cc

```diff
@@ -225,7 +225,7 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::Expression *what, BasicBlock
                     current->exprs.emplace_back(aliasName, a->loc, make_unique<Alias>(a->symbol));
                 }
 
-                current->exprs.emplace_back(cctx.target, a->loc, make_unique<Ident>(aliasName));
+                current->exprs.emplace_back(cctx.target, core::LocOffsets::none(), make_unique<Ident>(aliasName));
```

Comment:
I'd really love if we could make it so that we don't have to rely on `none` locs in too many places.

What do you think about trying something here:

https://github.com/sorbet/sorbet/blob/b7fdfb76c540f927aecb1f6741ac1bfa99b320ef/infer/environment.cc#L837-L840

where in addition to checking `lspQueryMatches`, we also check if the ident was "synthesized"?

I'd love to use "sythesized" as the bit for whether something should be ignored, because none Locs usually manifest in weird ways, like printing `???` in error messages and making things harder to debug.

Reply:
I like the idea of relying on synthesized much more than relying constructing the instruction with no location information. I'll experiment with threading through checks for synthesized instead.